velox/exec/tests/HashJoinTest.cpp

```diff
@@ -5177,6 +5177,109 @@ TEST_F(HashJoinTest, dynamicFiltersAppliedToPreloadedSplits) {
+        probeSplits.push_back(
+            exec::Split(makeHiveConnectorSplit(tempFiles.back()->getPath())));
+      }
+      SplitInput splits;
+      splits.emplace(nodeId, probeSplits);
+      return splits;
+    };
+  };
+
+  // Create build data
+  std::vector<RowVectorPtr> buildVectors;
+  for (int i = 0; i < 10; ++i) {
+    buildVectors.push_back(makeRowVector({
+        makeFlatVector<int32_t>(
+            numRowsBuild,
+            [i](auto row) { return 35 + 2 * (row + i * numRowsBuild / 5); }),
+        makeFlatVector<int64_t>(numRowsBuild, [](auto row) { return row; }),
+    }));
+  }
+
+  createDuckDbTable("t", probeVectors);
+  createDuckDbTable("u", buildVectors);
+
+  auto probeType = ROW({"c0", "c1"}, {INTEGER(), BIGINT()});
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  auto buildSide = PlanBuilder(planNodeIdGenerator, pool_.get())
+                       .values(buildVectors)
+                       .project({"c0 AS u_c0", "c1 AS u_c1"})
+                       .planNode();
+
+  // Inner join.
+  core::PlanNodeId probeScanId;
+  core::PlanNodeId joinId;
+  auto op = PlanBuilder(planNodeIdGenerator, pool_.get())
+                .tableScan(probeType)
+                .capturePlanNodeId(probeScanId)
+                .aggregation(
+                    {"c0", "c1"},
+                    {"sum(c1)"},
+                    {},
+                    core::AggregationNode::Step::kPartial,
+                    false)
+                .hashJoin(
+                    {"c0", "c1"},
+                    {"u_c0", "u_c1"},
+                    buildSide,
+                    "",
+                    {"c0", "c1", "a0"},
+                    core::JoinType::kInner)
+                .capturePlanNodeId(joinId)
+                .project({"c0", "c1", "a0"})
+                .planNode();
+
+  HashJoinBuilder(*pool_, duckDbQueryRunner_, driverExecutor_.get())
+      .planNode(std::move(op))
+      .makeInputSplits(makeInputSplits(probeScanId))
+      .referenceQuery(
+          "SELECT t.c0, t.c1, sum(t.c1) FROM t, u WHERE t.c0 = u.c0 and t.c1 = u.c1 group by t.c0, t.c1")
+      .verifier([&](const std::shared_ptr<Task>& task, bool hasSpill) {
+        SCOPED_TRACE(fmt::format("hasSpill:{}", hasSpill));
+        auto planStats = toPlanStats(task->taskStats());
+        if (hasSpill) {
+          // Dynamic filtering should be disabled with spilling triggered.
+          ASSERT_EQ(0, getFiltersProduced(task, 2).sum);
+          ASSERT_EQ(0, getFiltersAccepted(task, 0).sum);
+          ASSERT_EQ(getInputPositions(task, 1), numRowsProbe * numSplits);
+          ASSERT_TRUE(planStats.at(probeScanId).dynamicFilterStats.empty());
+        } else {
+          ASSERT_EQ(2, getFiltersProduced(task, 2).sum);
+          ASSERT_EQ(2, getFiltersAccepted(task, 0).sum);
+          ASSERT_EQ(0, getReplacedWithFilterRows(task, 1).sum);
+          ASSERT_LT(getInputPositions(task, 1), numRowsProbe * numSplits);
+          ASSERT_EQ(
```

Comment:
> planStats.at(probeScanId).dynamicFilterStats

Can you extract a temp variable for this and re-use it in both branches for readability?

Reply:
Done.